tests/test_new.py

```diff
@@ -0,0 +1,61 @@
+# test_new.py - I don't know what to call this yet.
```

Comment:
You had been tempted in the past to discard the old tests and write new ones. I haven't referred to the existing tests yet while using this feature except to check that they pass.

I will review the existing tests to see which ones are relevant to this PR and if any of them can be updated or replaced.